tests/ga/test_update.py

```diff
@@ -686,12 +686,13 @@ def _test_run_latest(self, mock_child=None, mock_time=None, child_args=None):
     def test_run_latest(self):
         self.prepare_agents()
 
-        agent = self.update_handler.get_latest_agent_greater_than_daemon()
-        args, kwargs = self._test_run_latest()
-        args = args[0]
-        cmds = textutil.safe_shlex_split(agent.get_agent_cmd())
-        if cmds[0].lower() == "python":
-            cmds[0] = sys.executable
+        with patch("azurelinuxagent.common.conf.get_autoupdate_enabled", return_value=True):
```

Comment:
This change as a result of removing global mock for autoupdate flag in AgentUpdateCase. See the comment in tools.py file

Reply:
do we know why we had a global mock?

Reply:
My theory after digging history of this change, this was added when agent update feature implemented first. That time they don't mock for update_handler and all of them extending agent setup they decided added to global setup. Now we have update_handler mock and the agent update is related to update_handler, I don't think we need a global mock and also causing problems for other tests like I mentioned before.